cpp/src/arrow/engine/substrait/serde.cc

```diff
@@ -86,6 +86,20 @@ Result<std::vector<compute::Declaration>> DeserializePlan(
   return sink_decls;
 }
 
+Result<compute::ExecPlan> DeserializePlan(const Buffer& buf,
+                                          const ConsumerFactory& consumer_factory,
+                                          ExtensionSet* ext_set_out) {
```

Comment:
I do like the name `ext_set_out` more than `ext_set`.  Could you update the name in the header as well so they match?

Reply:
Renamed ext_set to ext_set_out.